src/main/java/com/example/bookstore/service/user/UserServiceImpl.java

```diff
@@ -27,6 +29,7 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
         User user = userMapper.toUser(requestDto);
         user.setPassword(passwordEncoder.encode(user.getPassword()));
         userRepository.save(user);
+        shoppingCartService.getShoppingCartByUserId(user);
```

Comment:
```suggestion
        shoppingCartService.createShoppingCart(user);
```

Method should look like
```
@Override
    public void createShoppingCart(User user) {
        ShoppingCart shoppingCart = new ShoppingCart();
        shoppingCart.setUser(user);
        repository.save(shoppingCart);
    }
```